File: problems/535_encode_and_decode_tinyurl/Solution.java
import java.util.*;

public class Codec {
    Map<String, String> map = new HashMap<>();
    int counter = 0;
    
    public String encode(String longUrl) {
        String key = String.valueOf(++counter);
        map.put(key, longUrl);
        return "http://tinyurl.com/" + key;
    }
    
    public String decode(String shortUrl) {
        String key = shortUrl.substring(shortUrl.lastIndexOf('/') + 1);
        return map.get(key);
    }
}

File: problems/535_encode_and_decode_tinyurl/solution.py
class Codec:
    def __init__(self):
        self.map = {}
        self.counter = 0
    
    def encode(self, longUrl: str) -> str:
        self.counter += 1
        key = str(self.counter)
        self.map[key] = longUrl
        return 'http://tinyurl.com/' + key
    
    def decode(self, shortUrl: str) -> str:
        key = shortUrl.split('/')[-1]
        return self.map[key]
